pkg/controller/vsphere/machine_scope_test.go

```diff
@@ -279,8 +281,13 @@ func TestPatchMachine(t *testing.T) {
 	model, _, server := initSimulator(t)
 	defer model.Remove()
 	defer server.Close()
-	credentialsSecretUsername := fmt.Sprintf("%s.username", server.URL.Host)
-	credentialsSecretPassword := fmt.Sprintf("%s.password", server.URL.Host)
+	host, port, err := net.SplitHostPort(server.URL.Host)
```

Comment:
how is this change related to this PR?

Reply:
We need to pass the test provider config where insecure connection is allowed to all tests because simulator doesn't support secure connections.